From 1c3bf032ddd8119b31c3c936322b2a721b6cdd11 Mon Sep 17 00:00:00 2001
From: Cody Xie <cody.xie@rock-chips.com>
Date: Mon, 19 Nov 2018 12:01:19 +0800
Subject: [PATCH] arm64: dts: rockchip: Add l2 cache topology for rk3308

Commit 5d425c18653731af6 ("arm64: kernel: add support for cpu cache
information") adds cacheinfo support for ARM64. Since there's no
architectural way of detecting the cpus that share particular cache,
device tree can be used and the core cacheinfo already supports the
same.

This patch adds the l2 cache topology on RK3308 SoC based boards.

Change-Id: Iaa0ddb11d02ba9cd42aea59406e02160affd32bf
Signed-off-by: Cody Xie <cody.xie@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3308.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308.dtsi b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
index eb354157729d..3a98812f4d7a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
@@ -51,6 +51,7 @@
 			dynamic-power-coefficient = <90>;
 			operating-points-v2 = <&cpu0_opp_table>;
 			cpu-idle-states = <&CPU_SLEEP>;
+			next-level-cache = <&l2>;
 		};
 
 		cpu1: cpu@1 {
@@ -60,6 +61,7 @@
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			cpu-idle-states = <&CPU_SLEEP>;
+			next-level-cache = <&l2>;
 		};
 
 		cpu2: cpu@2 {
@@ -69,6 +71,7 @@
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			cpu-idle-states = <&CPU_SLEEP>;
+			next-level-cache = <&l2>;
 		};
 
 		cpu3: cpu@3 {
@@ -78,6 +81,7 @@
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			cpu-idle-states = <&CPU_SLEEP>;
+			next-level-cache = <&l2>;
 		};
 
 		idle-states {
@@ -92,6 +96,10 @@
 				min-residency-us = <900>;
 			};
 		};
+
+		l2: l2-cache {
+			compatible = "cache";
+		};
 	};
 
 	cpu0_opp_table: cpu0-opp-table {
-- 
2.35.3

